Lint/git plugins

tbhaxor opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Fixed the linting for `plugins/available/git.plugin.bash` file

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

tbhaxor commented:
@NoahGorny if you find green coloured issues via shellcheck, here is hack to fix all of them

```
shellcheck -f diff <filename> | git apply
```

Remember this as it saved majority of my time while fixing linting :stuck_out_tongue:

cornfeedhobo commented on plugins/available/git.plugin.bash:
```diff
@@ -1,275 +1,314 @@
 function git_remote {
-  about 'adds remote $GIT_HOSTING:$1 to current repo'
-  group 'git'
+	about "adds remote $GIT_HOSTING:$1 to current repo"
```
Interesting that this `about` was ever used. I don't think it would work as intended when looking at `glossary`, but I haven't tested.

cornfeedhobo reviewed (approved):
I'm approving this as just a lint PR. I assume any fixed can be followed up on in another PR, but I've left a comment for the future. Great work!

NoahGorny reviewed (approved):
I rebased and added a small change.
Well done @tbhaxor and thank you for the CR @cornfeedhobo

tbhaxor commented:
> I rebased and added a small change.
> Well done @tbhaxor and thank you for the CR @cornfeedhobo

Thanks man. I am occupied in project. I am not getting time to fix my PR's
